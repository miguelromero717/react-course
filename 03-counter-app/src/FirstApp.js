import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({ title, subtitle }) => {
    const sayHi = title;
    const object = {
        name: 'Miguel',
        age: 30
    };

    return (
        <>
            <h1>{sayHi}</h1>
            <pre>{JSON.stringify(object, null, 3)}</pre>
            <p>{subtitle}</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    subtitle: 'My first App'
}

export default FirstApp;