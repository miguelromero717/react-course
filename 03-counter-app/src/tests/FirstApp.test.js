import "@testing-library/jest-dom";
import FirstApp from "../FirstApp";
import { shallow } from 'enzyme';

describe('First App tests', () => {
    /*test('should shows message "Hola soy Miguel"', () => {
        const message = "Hola soy Miguel";
        const { getByText } = render(<FirstApp title={message}/>);

        expect(getByText(message)).toBeInTheDocument();
    })*/

    test('should shows FirstApp correctly', () => {

        const title = "Hola Soy Miguel";
        const wrapper = shallow(<FirstApp title={title} />);

        expect(wrapper).toMatchSnapshot();

    })

    test('should shows subtitle by props', () => {
        const title = "Hola Soy Miguel";
        const subtitle = "New subtitle";
        const wrapper = shallow(
            <FirstApp
                title={title}
                subtitle={subtitle}
            />
        );

        const text = wrapper.find('p').text();

        expect(text).toBe(subtitle);
    });


})
