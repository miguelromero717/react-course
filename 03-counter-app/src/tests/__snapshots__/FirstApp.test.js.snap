// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First App tests should shows FirstApp correctly 1`] = `
<Fragment>
  <h1>
    Hola Soy Miguel
  </h1>
  <p>
    My first App
  </p>
</Fragment>
`;
